test_guide.md

```diff
@@ -0,0 +1,137 @@
+# How to write a Node.js test
+
+## What is a test?
+
+A test must be a node script that exercises a specific functionality provided by node and
+checks that it behaves as expected. It should return 0 on success, otherwise it will
+fail. A test will fail if:
+
+- It exits by calling `process.exit(code)` where `code != 0`
+- It exits due to an uncaught exception.
+- It never exits. In this case, the test runner will terminate the test because it sets a maximum time limit.
+
+They can be added for multiple reasons:
+
+- When adding new functionality.
+- When fixing regression and bugs.
+- Expanding test coverage.
+
+
+## Test structure
+
+Let's analyze this very basic test from the Node.js test suite:
+
+```javascript
+1  'use strict'; 
+2  const common = require('../common');
+3  const http = require('http');
+4  const assert = require('assert');
+5
+6  const server = http.createServer(common.mustCall((req, res) => {
+7    res.end('ok');
+8  }));
+9  server.listen(common.PORT, () => {
+10   http.get({
+11     port: common.PORT,
+12     headers: {'Test': 'Düsseldorf'}
+13   }, common.mustCall((res) => {
+14     assert.equal(res.statusCode, 200);
+15     server.close();
+16   }));
+17 });
+```
+
+**Lines 1-2**
+
+```
+'use strict';
+const common = require('../common');
+```
+
+These two lines are mandatory and should be included on every test.
+The `common` module is a helper module that provides useful tools for the tests.
+If for some reason, no functionality from `common` is used, it should still be included like this:
+
+```
+require('../common');
+````
+
+Why? It checks for leaks of globals.
+
+**Lines 3-4**
+
+```
+const http = require('http');
+const assert = require('assert');
+```
+
+These modules are required for the test to run. Except for special cases, these modules should only include core modules.
+The `assert` module is used by most of the tests to check that the assumptions for the test are met.
```

Comment:
My opinion only, but `const` vs. `var` is a broader topic project-wide that doesn't need to necessarily be mentioned here. It *can* be, surely. But if there is a strong desire to make it as lean as possible, it can probably be left out just as well. I wouldn't hold up on publishing this over the issue either way.